src/components/CheckableInput/CheckableInput.tsx

```diff
@@ -0,0 +1,52 @@
+import classNames from "classnames";
+import { TSFixMe } from "index";
+import { nanoid } from "nanoid";
+import PropTypes from "prop-types";
+import React, { useEffect, useRef, HTMLProps } from "react";
+
+type Props = {
+  type: string;
+  label: string;
+  required?: boolean;
+  indeterminate?: boolean;
+} & HTMLProps<HTMLElement>;
+
+const CheckableInput = ({
+  type,
+  label,
+  required = false,
+  indeterminate = false,
+  ...checkboxProps
+}: Props): JSX.Element => {
+  const inputId = useRef(nanoid());
+  // setting this type as any for now, otherwise it causes linting errors
+  const inputRef: TSFixMe = useRef(null);
+
+  useEffect(() => {
+    inputRef.current.indeterminate = indeterminate;
```

Comment:
Can a radio button be indeterminate? If not sure this component stack makes much sense.

Reply:
Just talked this through with @bartaz - I see what you mean, but we don't define `indeterminate` as a prop on `RadioInput`, so it couldn't pass it if it wanted to, and I'm not sure how we could lift that `useEffect` block up into the `CheckboxInput` component.